Makefile

```diff
@@ -478,66 +477,48 @@ proto-all: proto-update-deps proto-format proto-lint proto-check-breaking proto-
-containerProtoGen=prov-proto-gen-$(containerProtoVer)
-containerProtoGenSwagger=prov-proto-gen-swagger-$(containerProtoVer)
-containerProtoFmt=prov-proto-fmt-$(containerProtoVer)
+protoVer=0.15.1
+protoImageName=ghcr.io/cosmos/proto-builder:$(protoVer)
+protoImage=$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName)
 
 # The proto gen stuff will update go.mod and go.sum in ways we don't want (due to docker stuff).
 # So we need to go mod tidy afterward, but it can't go in the scripts for the same reason that we need it.
 
 proto-gen:
 	@echo "Generating Protobuf files"
 	cp go.mod .go.mod.bak
-	if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoGen}$$"; then \
-		docker start -a $(containerProtoGen); \
-	else \
-		docker run --name $(containerProtoGen) -v $(CURDIR):/workspace --workdir /workspace $(containerProtoImage) \
-			sh ./scripts/protocgen.sh; \
-	fi
+	$(protoImage) sh ./scripts/protocgen.sh
 	mv .go.mod.bak go.mod
 	$(GO) mod tidy
 
 proto-swagger-gen:
 	@echo "Generating Protobuf Swagger"
-	if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoGenSwagger}$$"; then \
-		docker start -a $(containerProtoGenSwagger); \
-	else \
-		docker run --name $(containerProtoGenSwagger) -v $(CURDIR):/workspace --workdir /workspace $(containerProtoImage) \
-			sh ./scripts/protoc-swagger-gen.sh; \
-	fi
-	$(GO) mod tidy
+	$(protoImage) sh ./scripts/protoc-swagger-gen.sh
 
 proto-doc-gen:
 	@echo "Generating Protobuf Markdown"
 	./scripts/proto-doc-gen.sh
 
 proto-format:
 	@echo "Formatting Protobuf files"
-	if docker ps -a --format '{{.Names}}' | grep -Eq "^${containerProtoFmt}$$"; then \
-		docker start -a $(containerProtoFmt); \
-	else \
-		docker run --name $(containerProtoFmt) -v $(CURDIR):/workspace --workdir /workspace tendermintdev/docker-build-proto \
-			find . \
-				-not -path './third_party/*' \
-				-not -path './vendor/*' \
-				-not -path './protoBindings/*' \
-				-name '*.proto' \
-				-exec clang-format -i {} \; ; \
-	fi
+	$(protoImage) find . \
+			-not -path './third_party/*' \
+			-not -path './vendor/*' \
+			-not -path './protoBindings/*' \
+			-name '*.proto' \
+			-exec clang-format -i {} \; ; \
 
 proto-lint:
 	@echo "Linting Protobuf files"
-	$(DOCKER_BUF) lint --error-format=json
+	$(protoImage) buf lint --error-format=json
 
 proto-check-breaking:
 	@echo "Check breaking Protobuf files"
-	$(DOCKER_BUF) breaking proto --against '$(HTTPS_GIT)#branch=main,subdir=proto' --error-format=json
+	$(protoImage) buf breaking proto --against '$(HTTPS_GIT)#branch=main,subdir=proto' --error-format=json
 
 proto-check-breaking-third-party:
 	@echo "Check breaking 3rd party Protobuf files"
-	$(DOCKER_BUF) breaking third_party/proto --against '$(HTTPS_GIT)#branch=main,subdir=third_party/proto' --error-format=json
+	$(protoImage) buf breaking third_party/proto --against '$(HTTPS_GIT)#branch=main,subdir=third_party/proto' --error-format=json
```

Comment:
If we compare against a stable version, then if one PR has a breaking change and is merged, the check will fail on all subsequent PRs until we tag a new version. So this check would then become useless; we'd learn to just ignore it since it would quite often be expected to fail. And if we expect it to fail due to a previous PR, we're not going to notice if our PR causes a breaking change that we **DO** need to worry about.

Reply:
Oops. I have a slight clarification here.

In my previous comment, I was thinking more about the GitHub action we have that run a buf breaking check. That action does not use this make target. That action runs the check agains the PRs target branch (which is usually `main` but is sometimes a `release/va.b.x` branch).

 But my argument still holds for this make target. If we're always comparing back to a specific stable tag or commit, then, if this check fails, it's not clear whether it's because of changes that are in my current branch or changes that were merged into `main` since the stable tag or commit was created (assuming I've based my branch off of `main`). By comparing back to `main` here, then, if the check fails, we know it's because of something we've done in our current branch, which is exactly what we want out of this check.